src/openrct2/object/Object.cpp

```diff
@@ -329,6 +329,48 @@ std::unique_ptr<IStream> ObjectAsset::GetStream() const
     return {};
 }
 
+u8string VersionString(const ObjectVersion& version)
+{
+    return std::to_string(std::get<0>(version)) + "." + std::to_string(std::get<1>(version)) + "."
+        + std::to_string(std::get<2>(version));
+}
+
+ObjectVersion VersionTuple(std::string_view version)
+{
+    if (version.empty())
+    {
+        return std::make_tuple(0, 0, 0);
+    }
+
+    auto nums = String::Split(version, ".");
+    uint16_t versions[VersionNumFields] = {};
+    if (nums.size() > VersionNumFields)
+    {
+        log_warning("%i fields found in version string '%s', expected X.Y.Z", nums.size(), version);
+    }
+    if (nums.size() == 0)
+    {
+        log_warning("No fields found in version string '%s', expected X.Y.Z", version);
+        return std::make_tuple(0, 0, 0);
+    }
+    // fill in array from right-to-left
+    try
+    {
+        uint8_t highestIndex = std::min(static_cast<int8_t>(nums.size()), VersionNumFields);
+        for (int8_t i = highestIndex - 1; i >= 0; i--)
+        {
+            uint16_t value = stoi(nums.at(i));
```

Comment:
stoi returns int, this will truncate if someone uses a value higher than 65535, I'm actually a bit surprised there is no compiler warning generated here, its probably best to first read it as int and validate if the version component fits uint16_t and warn the user if it will truncate.